src/lucky/tags/with_defaults.cr

```diff
@@ -0,0 +1,73 @@
+module Lucky::WithDefaults
+  def with_defaults(**named_args)
+    OptionMerger.new(context: self, named_args: named_args).run do |html|
+      yield html
+    end
+  end
+
+  class OptionMerger(T, V)
+    def initialize(@context : T, @named_args : V)
+    end
+
+    def run
+      yield self
+    end
+
+    macro method_missing(call)
+      overridden_html_class = nil
+
+      {% named_args = call.named_args %}
+      {% if named_args %}
+        {% if call.named_args.any? { |arg| arg.name == :class } %}
+          {% raise <<-ERROR
+
+
+          Use 'replace_class' or 'append_class' instead of 'class'.
+
+          Correct example:
+
+              with_defaults class: "default" do |html|
+                # Use 'replace_class' or 'append_class' here
+                html.div replace_class: "replaced"
+              end
+
+          Incorrect example:
+
+              with_defaults class: "default" do |html|
+                # Won't work with 'class'
+                html.div class: "replaced"
+              end
+
+          -----------------
+
+          ERROR
+          %}
+        {% end %}
+
+        {% appended_class_arg = call.named_args.find { |arg| arg.name == :append_class } %}
+        {% if appended_class_arg %}
+          overridden_html_class = "#{@named_args[:class]?} #{{{ appended_class_arg.value }}}"
+        {% end %}
+        {% named_args = named_args.reject { |arg| arg.name == :append_class } %}
+
+        {% replace_class_arg = call.named_args.find { |arg| arg.name == :replace_class } %}
+        {% if replace_class_arg %}
+          overridden_html_class = "#{{{ replace_class_arg.value }}}"
+        {% end %}
+        {% named_args = named_args.reject { |arg| arg.name == :replace_class } %}
```

Comment:
This is really unpleasant. I'll try to find a way to clean this up. I also need to see how this affects compile time performance

Reply:
Performance can probably be improved but I think it is fine. 5000 calls with `with_defaults` and it added 1-2 seconds. This will probably be used sparingly (mostly for form inputs) so I think this is pretty reasonable